### PR TITLE
[Feature][e2e] Suggest e2e test adapt M1 chip

### LINKED ISSUE #9039: [Feature][e2e] Suggest e2e test adapt M1 chip
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Currently ```E2E``` test does not support running in M1 chip.

Suggest to support it.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

This pr will close #9039.

Fix ```WorkFlowE2ETest``` error in ```https://github.com/apache/dolphinscheduler/runs/5638351689?check_suite_focus=true```.

### CODE REVIEW

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java [hunk 1]
```diff
@@ -59,12 +59,13 @@
...
 
 import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.utility.DockerImageName;
 
 @Slf4j
-final class DolphinSchedulerExtension
-    implements BeforeAllCallback, AfterAllCallback,
-    BeforeEachCallback {
-    private final boolean LOCAL_MODE = Objects.equals(System.getProperty("local"), "true");
+final class DolphinSchedulerExtension implements BeforeAllCallback, AfterAllCallback, BeforeEachCallback {
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java:65]: Please try to avoid these unnecessary changes

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java [hunk 2]
```diff
@@ -59,12 +59,13 @@
...
+    private final boolean LOCAL_MODE = Objects.equals(System.getenv("local"), "true");
+
+    private final boolean M1_CHIP_FLAG = Objects.equals(System.getenv("m1_chip"), "true");
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: We are using system properties, not environment variables

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java [hunk 3]
```diff
@@ -73,36 +74,27 @@
     private HostAndPort address;
     private String rootPath;
 
+    private DockerImageName imageName;
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: This can be a local variable, it doesn't need to be a field

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java [hunk 4]
```diff
@@ -73,36 +74,27 @@
...
-            record = Paths.get(System.getenv("RECORDING_PATH"));
-            if (!record.toFile().exists()) {
-                if (!record.toFile().mkdir()) {
-                    throw new IOException("Failed to create recording directory: " + record.toAbsolutePath());
-                }
-            }
-        } else {
-            record = Files.createTempDirectory("record-");
-        }
+        getBrowserContainerByOsName();
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: `getXxxx` usually returns an object and is assigned to some variable, but your  `getXxxx` instead, sets a field, which I don't think is a good practice.

Please either change the method name to `setXxx` or make it return the object and assign it to the field

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java [hunk 5]
```diff
@@ -159,6 +151,39 @@ public Statement apply(Statement base, Description description) {
         rootPath = "/dolphinscheduler/ui/";
     }
 
+    private void getBrowserContainerByOsName() {
+        if (LOCAL_MODE && M1_CHIP_FLAG) {
+            imageName = DockerImageName.parse("seleniarm/standalone-chromium:4.1.2-20220227")
```
Comment (kezhenxu94) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java:155]: If this image also works for non-M1, I'd just upgrade the image for all systems, not just M1

Comment (SbloodyS) [reply dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java:155]: This image only work for ARM64 chip.

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java (hunk 4 above)
Comment (SbloodyS) [reply dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: I'll fix it.

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java (hunk 2 above)
Comment (SbloodyS) [reply dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: I'll fix it.

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java (hunk 3 above)
Comment (SbloodyS) [reply dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: I'll fix it.

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java (hunk 1 above)
Comment (kezhenxu94) [reply dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java:65]: This is still not avoided, please pay attention next time

**Review by kezhenxu94 — APPROVED** [review]
